jetbrains-core/tst/software/aws/toolkits/jetbrains/services/cloudwatch/logs/CloudWatchLogStreamCoroutineTest.kt

```diff
@@ -0,0 +1,165 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.cloudwatch.logs
+
+import com.intellij.testFramework.ProjectRule
+import com.intellij.ui.table.TableView
+import com.intellij.util.ui.ListTableModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.ClosedSendChannelException
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.withTimeout
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.assertThatThrownBy
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito
+import org.mockito.Mockito.`when`
+import software.amazon.awssdk.services.cloudwatchlogs.CloudWatchLogsAsyncClient
+import software.amazon.awssdk.services.cloudwatchlogs.model.GetLogEventsRequest
+import software.amazon.awssdk.services.cloudwatchlogs.model.GetLogEventsResponse
+import software.amazon.awssdk.services.cloudwatchlogs.model.OutputLogEvent
+import software.aws.toolkits.jetbrains.core.MockClientManagerRule
+import software.aws.toolkits.resources.message
+import java.util.concurrent.CompletableFuture
+
+class CloudWatchLogStreamCoroutineTest {
+    @JvmField
+    @Rule
+    val projectRule = ProjectRule()
+
+    @JvmField
+    @Rule
+    val mockClientManagerRule = MockClientManagerRule(projectRule)
+
+    private var testCoroutineScope: CoroutineScope = TestCoroutineScope()
+
+    @Before
+    fun before() {
+        testCoroutineScope = TestCoroutineScope()
+    }
+
+    @After
+    fun after() {
+        try {
+            testCoroutineScope?.cancel()
+        } catch (e: Exception) {
+        }
+    }
+
+    @Test
+    fun modelIsPopulated() {
+        val mockClient = mockClientManagerRule.create<CloudWatchLogsAsyncClient>()
+        `when`(mockClient.getLogEvents(Mockito.any<GetLogEventsRequest>()))
+            .thenReturn(CompletableFuture.completedFuture(GetLogEventsResponse.builder().events(OutputLogEvent.builder().message("message").build()).build()))
+        val tableModel = ListTableModel<OutputLogEvent>()
+        val table = TableView<OutputLogEvent>(tableModel)
+        val coroutine = CloudWatchLogStreamCoroutine(mockClient, table, "abc", "def")
+        runBlocking {
```

Comment:
should use `runBlockingTest`

Reply:
What we are testing doesn't use dealy and runBlockingTest is doing this: https://github.com/Kotlin/kotlinx.coroutines/issues/1204 so runBlocking is fine for now